radixdlt/src/main/java/com/radixdlt/consensus/SyncQueues.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.consensus;
+
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableMap;
+import com.radixdlt.counters.SystemCounters;
+import com.radixdlt.counters.SystemCounters.CounterType;
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.crypto.Hash;
+import io.reactivex.rxjava3.annotations.Nullable;
+import java.util.LinkedList;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Synchronous queuing mechanism for consensus events which require syncing
+ * before being able to effectively process it.
+ *
+ * A separate queue is created for each node in order to keep the message ordering invariant.
```

Comment:
Would be good to remark that this is not thread safe (due to the use of plain `LinkedLists` in `SyncQueue`)